core/src/main/java/org/testcontainers/utility/ResourceReaper.java

```diff
@@ -157,7 +159,14 @@ public InspectContainerResponse getContainerInfo() {
                     while (true) {
                         RYUK_ACK_RATE_LIMITER.doWhenReady(() -> {
                             int index = 0;
-                            try(Socket clientSocket = new Socket(host, ryukPort)) {
+                            try(Socket clientSocket = new Socket()) {
+                                int socketReadTimeOut = TestcontainersConfiguration.getInstance().getRyukTimeout() * 1000;
+                                int socketConnectTimeOut = 5 * 1000;
+                                clientSocket.setSoTimeout(socketReadTimeOut);
+                                Unreliables.retryUntilTrue(socketConnectTimeOut, TimeUnit.MILLISECONDS, () -> {
```

Comment:
this timeout should be higher than the inner one, otherwise there will be no retries at all

Reply:
Also, since there is a `catch` statement for `IOException`, perhaps we can rely on it, and make the whole operation of creating the Socket retried, making `Unreliables` here unnecessary

Reply:
Should I keep the retry for the whole creating socket operation for 5 seconds?

Reply:
since this is a rare and exceptional situation, I would say there is no need to, as long as the whole thing is retried 👍